src/neuralnet/neuralnet.cpp

```diff
@@ -0,0 +1,32 @@
+#include "neuralnet.h"
+#include "neuralnet_stub.h"
+
+#if defined(QT_GUI) && defined(WIN32)
+#include "neuralnet_win.h"
+#endif
+
+using namespace NN;
+
+namespace
+{
+    INeuralNetPtr instance;
+}
+
+INeuralNetPtr NN::CreateNeuralNet()
+{
```

Comment:
With the new scraper NN the body will look something like this:

```
if(new scraper enabled)
   return NewScraperNN

// Fallback to default implementations
#ifdef WIN32
 return Win32NN
#else
 return StubNN
#endif
```

The ifdef is still needed since the Win32 NN calls Qt. We can get around this by injecting a `DotNetExecutor` or `Bridge` object which does the actual call. If you instantiate from Qt it will inject a QAxObject capable executor, otherwise it injects a dummy executor. That will also allow us to stop making NN implementation selections at compile time.